### PR TITLE
fix(stock): don't reserve qty on sales return.

### PR DESCRIPTION
**Pull request by dj12djdjs**
bringing https://github.com/frappe/erpnext/pull/31017 back to life

### CODE REVIEW

**Comment by dj12djdjs** [comment]
@ankush Just wanted to checkup, have you thought of any way to implement this without disturbing existing workflows ?

**Comment by ankush** [comment]
@dj12djdjs we can add an option in selling settings. 

The default behavior should be consistent with current behavior.

**Comment by dj12djdjs** [comment]
@ankush Ready for review

**Comment by dj12djdjs** [comment]
@ankush Sorry about that, I just realized there was more conflicts 😅

**Comment by dj12djdjs** [comment]
@ankush Is there any issues with this ?

**Review by s-aga-r — CHANGES_REQUESTED** [review]
@dj12djdjs please resolve conflicts.

**Review by barredterra — CHANGES_REQUESTED** [review]
Changes to `erpnext/selling/doctype/sales_order/sales_order.json` seem to be unrelated; please revert these.

File: erpnext/stock/stock_balance.py [hunk 1]
```diff
@@ -94,51 +98,85 @@ def get_balance_qty_from_sle(item_code, warehouse):
 
 
 def get_reserved_qty(item_code, warehouse):
```
Comment (rohitwaghchaure) [inline erpnext/stock/stock_balance.py:96]: This function looks very complex and difficult to read, could you split the code into multiple methods?

Comment (dj12djdjs) [reply erpnext/stock/stock_balance.py:96]: @rohitwaghchaure Honestly, I don't even fully understand the query. I just reverted it back to the previous query and used f string to inject the condition.

**Review by rohitwaghchaure — APPROVED** [review]

**Pull request merged**